src/ppx/application.cpp

```diff
@@ -1682,6 +1682,9 @@ Result Application::CreateShader(const std::filesystem::path& baseDir, const std
 
 float Application::GetElapsedSeconds() const
 {
+    if (mStandardOptions.deterministic) {
+        return static_cast<float>(mFrameCount * (1.f / 60.f));
```

Comment:
I think this may end up non-deterministic across machines due to different rounding modes or other FP flags - but I think a better solution can wait.

Reply:
Do you mean if we recompile, or the same binary run on 2 machines?
I would be surprised if running the same binary on 2 machines yields different rounding operations. Isn't float rounding spec'd?

Reply:
(goal here is for multiple runs of the same binary to be deterministic, not across builds)

Reply:
Rounding mode can be set both at runtime (https://en.cppreference.com/w/c/numeric/fenv/feround) or at compile time (with compiler flags). I don't know if there are more knobs at the system level that can play a role. I don't know how much of a problem this is but it's worth keeping in mind as a potential source of non-determinism.

Reply:
If the goal is same binary on the same system, we are good. Same binary on different systems may be a problem yet uncommon I bet (e.g. a library calling the rounding mode setter or a system-level knob, if it exists).

Reply:
> Isn't float rounding spec'd?

That depends on the compile time options that are used. If you are using any form of fast math, the compiler could change the order of operations, to give different results. Even on the same machine, clang and gcc could return different results.

However, I would not worry about that for now.

Reply:
> That depends on the compile time options that are used.

That for sure, but I meant if x86's divss & cvttss2si producing the same result on AMD or M1 through rosetta by ex. But yes, agree it's not that important for now.

> Rounding mode can be set both at runtime (https://en.cppreference.com/w/c/numeric/fenv/feround) 

Oh didn't knew that! thanks!